airflow/upgrade/rules/pod_template_file_rule.py

```diff
@@ -31,9 +32,15 @@ class PodTemplateFileRule(BaseRule):
 """
 
     def check(self):
+        executor = conf.get('core', 'EXECUTOR')
+        if executor != Executors.KubernetesExecutor:
+            return
+
         pod_template_file = conf.get("kubernetes", "pod_template_file", fallback=None)
-        if not pod_template_file:
-            return (
-                "Please create a pod_template_file by running `airflow generate_pod_template`.\n"
-                "This will generate a pod using your aiflow.cfg settings"
-            )
+        if pod_template_file:
+            return
+
+        return (
+            "Please create a pod_template_file by running `airflow generate_pod_template`.\n"
+            "This will generate a pod using your aiflow.cfg settings"
```

Comment:
```suggestion
            "Please create a pod_template_file by running `airflow generate_pod_template`.\n"
            "This will generate a pod using your airflow.cfg settings"
```